### DISCUSSION TITLE
calc 2.14.0.12 release

### DISCUSSION THREAD

**Discussion by lcn2** [lcn2/calc#41] [Announcements]
The following are the changes in this release:

    Fixed a number of typos.

    Drop support for SunOS, IRIX and MINGW32_NT-5.0 targets.
    Drop support for CLOCK_SGI_CYCLE.  Drop testing for __MSDOS__.

    Minor improvement of various help files.  Made format of help
    files more consistent.

    Corrected a few comments in zrandom.c, including a case where
    the comment referred to 1007 when it should have used 2^32.

    Improved seed() generation.  Improved some comments in seed.c.
    Added have_environ.c to build have_environ.h in order to
    determine if:

	extern char **environ;	/* user environment */

    is an valid external symbol.

    Fixed documentation that referred to the old additive 55
    generator.  We have been using the subtractive 100 in place
    of the additive 55 generator for a while now.

    Fixed depend rule for custom/Makefile.

    Fixed how Makefile variable SHELL is set on macOS vs. Linux.
    Using /bin/sh on macOS due to how zsh treats globs by default.

    Added charbit.h to define CALC_CHARBIT which is either CHAR_BIT
    from <limits.h> or 8 or a value set by the Makefile variable
    CALC_CHARBIT.  Added have_limits.h to determine if <limits.h>
    is a system include file.

    Compiling now tests for _WIN64 as well when testing for _WIN32.

    Now assuming that <stdio.h> is available under _WIN32 and _WIN64.

    Added some preliminary notes about attempts to compile calc
    under Windows 11.

	BTW: While we are unable to use Windows 11, we welcome
	     Windows 11 developers to try compiling calc natively
	     (instead of via a Linux virtual machine).  If you are
	     able to compile Windows 11 natively, we would welcome
	     GitHub pull requests showing any needed modifications:

		https://github.com/lcn2/calc/pulls

	     Please also add notes to the 'Compiling calc under Windows 11'
	     section in README.WINDOWS file.

    We added Makefile.simple as part of the master branch source
    to help those who may be using a make tool that does not support
    GNU Make-like features such as "ifeq" and ":=".

    Fixed incorrect ustat_dev member ref in seed() builtin that
    was part of the unreleased calc v2.12.0.11.

<hr /><em>This discussion was created from the release <a href='https://github.com/lcn2/calc/releases/tag/v2.14.0.12'>calc 2.14.0.12 release</a>.</em>

**Comment by lcn2** [comment]
A few notes on near term calc development and release plans:

Interest has been expressed to make calc able to compile natively under Windows 11.
We do not have a Windows 11 platform to develop on, nor do we plan to obtain one. 
Instead, we are depending on the efforts of Windows 11 developers: providing them
with calc related guidance as needed. 

If the Windows 11 platform changes are well contained and do NOT negatively
impact other platforms, we plan to make a trial v2.14.0.13 release with changes
that shouldn't impact non-Windows platforms.  If that trial release goes well,
then we plan to issue a new v2.14.1.0 stable release.  While we don't anticipate
any problems with the method outlined in README.WINDOWS, if there are cross
platform problems, we would revert trial v2.14.0.13 changes.  Either way the
README.WINDOWS will be updated with guidance in a new v2.14.1.0 stable release.

=-=

There is an effort to add "Feature request: support hex floats #14" and "calc script debugging". However,
and "Feature request: support hex floats #14" would appear in 2.14.1.x release AFTER the 2.14.0.1 stable release.
The "calc script debugging" would appear in a 2.15.x.y release AFTER the 2.14.0.1 stable release.

=-=

We are also thinking about a calc DMG release process for macOS.

=-=

And of course, all the above plans are subject to change by impacts of reality. :-)

Comments and suggestions welcome.

Best wishes for the holidays and the new year: "Share and enjoy!" :-)

chongo (Landon Curt Noll) /\oo/\

**Reply by pmetzger** [reply]
One biggie for me will be to enable parallel builds, since downstream packager build machines are now all many-core...

**Comment by lcn2** [comment]
Noted, thanks.

**Comment by lcn2** [comment]
BTW @pmetzger, The changes we are now making to address the missing file release bug will make it easier for us to support parallel builds.
